pilot/pkg/model/push_context.go

```diff
@@ -793,6 +783,24 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
+func (ps *PushContext) GetHostsFromGateways(proxy *Proxy) sets.Set {
+	hostsFromGateways := sets.New()
+	for _, gw := range proxy.MergedGateway.GatewayNameForServer {
+		for _, vsConfig := range ps.VirtualServicesForGateway(proxy.ConfigNamespace, gw) {
+			vs, ok := vsConfig.Spec.(*networking.VirtualService)
+			if !ok { // should never happen
+				log.Errorf("Failed in getting a virtual service: %v", vsConfig.Labels)
```

Comment:
I would not handle this and let it panic if it does - it's a programming bug if so and would fail immediately and be detected